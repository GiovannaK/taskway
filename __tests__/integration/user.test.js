const {
  CREATE_USER,
  CONFIRM_ACCOUNT,
  LOGIN_USER,
  FORGOT_PASSWORD,
  RESET_PASSWORD} = require('../graphqlQueries/userQueries')
const server = require('../../src/server');
const {createTestClient} = require('apollo-server-testing')
const truncate = require('../utils/truncate')
const {User} = require('../../src/models')
const bcrypt = require('bcryptjs')

describe('Authentication flux', () => {
  beforeEach(async () => {
    await truncate();
  })
  it('Should register user and generate emailConfirmationToken', async () => {
    const serverTest = server
    const {mutate} = createTestClient(serverTest);

    const res = await mutate({
      mutation: CREATE_USER,
      variables: {
        firstName: "mytest",
        lastName: "lastnametest",
        email: "kowesor320@vvaa1.com",
        password: "testando"
      }
    })

    expect(res).toMatchSnapshot();
  })

  it('Should confirm user account', async () => {
    const password = "testando"

    const encryptedPassword = bcrypt.hashSync(password, 6)

    const user = await User.create({
      firstName: "teste1",
      lastName: "teste last",
      email: "teste@email.com",
      password: encryptedPassword,
      emailConfirmationToken: 'a783hd842fk3209akdfe09430',
      emailConfirmationExpires: Date.now() + 10 * (60 * 1000)
    })
    const serverTest = server
    const {mutate} = createTestClient(serverTest);

    const res = await mutate({
      mutation: CONFIRM_ACCOUNT,
      variables: {
        emailConfirmationToken: user.emailConfirmationToken,
      }
    })

    expect(res).toMatchSnapshot();
  })

  it('Should authenticate user with valid credentials', async () => {
    const password = "testando"

    const encryptedPassword = bcrypt.hashSync(password, 6)

    const user = await User.create({
      firstName: "teste1",
      lastName: "teste last",
      email: "teste@email.com",
      password: encryptedPassword,
      isVerified: true,
    })
    const serverTest = server
    const {mutate} = createTestClient(serverTest);

    const res = await mutate({
      mutation: LOGIN_USER,
      variables: {
        email: user.email,
        password: "testando"
      }
    })

    expect(res).toMatchSnapshot();
  })


  it('Should generate reset token', async () => {
    const password = "testando"

    const encryptedPassword = bcrypt.hashSync(password, 6)

    const user = await User.create({
      firstName: "teste1",
      lastName: "teste last",
      email: "teste@email.com",
      password: encryptedPassword,
      isVerified: true,
    })
    const serverTest = server
    const {mutate} = createTestClient(serverTest);

    const res = await mutate({
      mutation: FORGOT_PASSWORD,
      variables: {
        email: user.email,
      }
    })

    expect(res).toMatchSnapshot();
  })

  it('Should reset user password', async () => {
    const password = "testando"

    const encryptedPassword = bcrypt.hashSync(password, 6)

    const user = await User.create({
      firstName: "teste1",
      lastName: "teste last",
      email: "teste@email.com",
      password: encryptedPassword,
      isVerified: true,
      passwordResetToken: "are8tr8err9fjkmfjmgj9054",
      passwordResetExpires: Date.now() + 10 * (60 * 1000)
    })
    const serverTest = server
    const {mutate} = createTestClient(serverTest);

    const res = await mutate({
      mutation: RESET_PASSWORD,
      variables: {
        passwordResetToken: user.passwordResetToken,
        password: "testando2"
      }
    })

    expect(res).toMatchSnapshot();
  })
})