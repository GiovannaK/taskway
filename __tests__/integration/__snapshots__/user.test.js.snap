// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication flux Should authenticate user with valid credentials 1`] = `
Object {
  "data": Object {
    "userLogin": Object {
      "id": "970ecbc6-4c40-4877-aec3-c046d6eb0fa1",
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6Ijk3MGVjYmM2LTRjNDAtNDg3Ny1hZWMzLWMwNDZkNmViMGZhMSIsImlhdCI6MTYyMzcyNjU4MiwiZXhwIjoxNjIzODEyOTgyfQ.Y5lQPhh4CcNFEncGLcZ5UG6US9eol2gw3fpiwfI9Wak",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Authentication flux Should confirm user account 1`] = `
Object {
  "data": Object {
    "userConfirmAccount": Object {
      "email": "teste@email.com",
      "id": "2ffda1f8-9f2d-4e52-89b6-547509a5490e",
      "isVerified": true,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Authentication flux Should generate reset token 1`] = `
Object {
  "data": Object {
    "userForgotPassword": Object {
      "id": "eac51f41-719b-4e1a-bea1-68613a82e952",
      "passwordResetToken": "1cf614560833306adcdb1bad756f266050e2dffe",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Authentication flux Should register user and generate emailConfirmationToken 1`] = `
Object {
  "data": Object {
    "userRegister": Object {
      "email": "kowesor320@vvaa1.com",
      "emailConfirmationToken": "046374e569843b2890185481a3bbbea5c073cd5c",
      "firstName": "mytest",
      "id": "556b1d81-f4da-430a-a6de-fc707d538973",
      "lastName": "lastnametest",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Authentication flux Should reset user password 1`] = `
Object {
  "data": Object {
    "userResetPassword": Object {
      "email": "teste@email.com",
      "id": "68a88254-d34d-4c46-87de-fe8c1b002c5d",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
